ProjectManager/ProjectManager/App/Source/TaskForm/TaskFormViewModel.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  TaskFormViewModel.swift
+//  ProjectManager
+//
+//  Created by Harry, KokkiLE on 2023/05/22.
+//
+
+import Foundation
+import Combine
+
+final class TaskFormViewModel {
+    private let taskManager = TaskManager.shared
+    private var task: Task?
+    @Published var isEditable: Bool
+    
+    var title: String {
+        return task?.title ?? ""
+    }
+    
+    var body: String {
+        return task?.body ?? ""
+    }
+    
+    var deadline: Date {
+        return task?.deadline ?? Date()
+    }
+    
+    var leftBarButtonTitle: String {
+        return task != nil ? "Edit" : "Cancel"
+    }
+    
+    var rightBarButtonTitle: String {
+        return "Done"
+    }
+    
+    var navigationTitle: String? {
+        return task != nil ? task?.state.description : State.todo.description
+    }
+    
+    init(task: Task? = nil) {
+        self.task = task
+        self.isEditable = (task == nil)
+    }
+
+    func dismissOrEditableIfNeeded(action: (() -> Void)?) {
+        if task == nil {
+            action?()
+            
+            return
+        }
+        
+        self.isEditable = true
+    }
+    
+    func doneAction(title: String, date: Date, body: String) {
+        if task != nil {
+            updateTask(title: title, date: date, body: body)
+            
+            return
+        }
+        
+        if title != "" && body != "" {
+            addTask(title: title, date: date, body: body)
+        }
```

Comment:
```suggestion
        if task != nil {
            updateTask(title: title, date: date, body: body)
        } else if title != "" && body != "" {
            addTask(title: title, date: date, body: body)
        }
```
이게 좀 더 명확하지 않을까 싶긴한데요!

Reply:
넵, 저런식으로 하는 조건 검사도 좋을 것 같습니다🫡

`else if` 의 조건식은 RightBarbutton의 활성/비활성을 체크함에따라 필요없게 되어 제거하였습니다 !